### PR TITLE
feat: one-time checkout and supporting functions MARS-546

### PR DESCRIPTION
**Pull request by emuvente**
One-time async checkout, with some basket state and validation functions. Still needs to actually be tried out with an app, more tests, and a solution for tracking.

Despite the roughness, the basic flow should be all set here. Please let me know if anything looks off with the flow as it is now or if there are any missing pieces not already called out with a TODO!

### CODE REVIEW

**Review by mcstover — APPROVED** [review]
Great advancement of kv-shop methods and utils!

**Review by mcstover — APPROVED** [review]

File: @kiva/kv-shop/src/useBraintreeDropIn.ts [hunk 1]
```diff
@@ -44,7 +45,14 @@ export interface DropInInitOptions {
...
 
-export default function useBraintreeDropIn() {
+export interface DropInWrapper {
+	initDropIn: (options: DropInInitOptions) => Promise<Dropin>,
+	paymentMethodRequestable: Ref<boolean>,
+	requestPaymentMethod: () => Promise<PaymentMethodPayload | void>,
+	updateAmount: (amount: string|number) => void,
+}
+
+function initBraintreeDropin(): DropInWrapper {
```
Comment (dyersituations) [inline @kiva/kv-shop/src/useBraintreeDropIn.ts:55]: I can't comment on the line, but GitHub linting is saying that `event` is never used on line 95.

**Review by dyersituations — APPROVED** [review]

File: @kiva/kv-shop/src/useBraintreeDropIn.ts (hunk 1 above)
Comment (emuvente) [reply @kiva/kv-shop/src/useBraintreeDropIn.ts:55]: We do have a `// eslint-disable-next-line no-unused-vars` comment on it, but maybe github doesn't evaluate those in-line linting rules?

Comment (dyersituations) [reply @kiva/kv-shop/src/useBraintreeDropIn.ts:55]: Not sure, haven't spent much time with GitHub's linter.

**Pull request merged**